### PR TITLE
Update readme

### LINKED ISSUE #131: non-rolling update, i.e. kill pod before starting new
I have a pod that depends on a non-sharable volume.

Wave starts a new pod which never comes up because of this.

I'd like an annotation where the deployment is marked for a non-rolling update.

Possible?  New feature?

### PR DESCRIPTION
**Pull request by jabdoa2**
- Add compatibility matrix
- Notes on production setups
- Describe webhooks
- Remove finalizer/ownerReferences section
- Describe update strategies (fixes #131 by documenting how to do it)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 # Wave
 
-Wave watches Deployments within a Kubernetes cluster and ensures that each
-Deployment's Pods always have up to date configuration.
+Wave watches Deployments, StatefulSets and DaemonSets within a Kubernetes
+cluster and ensures that their Pods always have up to date configuration.
 
-By monitoring ConfigMaps and Secrets mounted by a Deployment, Wave can trigger
+By monitoring mounted ConfigMaps and Secrets mounted, Wave can trigger
```
Comment (toelke) [inline README.md:10]:
```suggestion
By monitoring mounted ConfigMaps and Secrets, Wave can trigger
```

File: README.md [hunk 2]
```diff
@@ -41,6 +41,22 @@ matches the live configuration.
...
+the previous and the next Kubernetes version.
+However, since Wave only edits Deployments, Daemonsets and StatefulSet
+we can support older Kubernetes verions as long as no fields were removed
+from those three objects.
+You can find supported versions in the following table:
+
+| Wave Version | API Client | Maximum Supported Kubernetes Versions | E2E Tested Versions |
+|--------------|------------|---------------------------------------|---------------------|
+| 0.5          | 1.14       | 1.15                                  |                     |
+| 0.6+         | 1.29       | 1.30                                  | 1.21, 1.29          |
```
Comment (toelke) [inline README.md:56]: We should just add 1.30 to the e2e-test (replacing 1.29 or in addition to).

File: hack/production.yaml [hunk 3]
```diff
@@ -10,7 +10,14 @@ pdb:
 topologySpreadConstraints:
   - maxSkew: 1
     topologyKey: topology.kubernetes.io/zone
-    whenUnsatisfiable: ScheduleAnyway
+    whenUnsatisfiable: DoNotSchedule
```
Comment (toelke) [inline hack/production.yaml:13]: I guess this fails in the 1-node test-cluster in the e2e tests now.

Comment (jabdoa2) [reply hack/production.yaml:13]: It should not. TopologySpread only assumes an even distribution. Evenly distributing to only one available node should work just fine ;-). But lets see.

Comment (jabdoa2) [reply hack/production.yaml:13]: Looks like you are right. Did not expect that to happen. Looks like minikube and kind are is missing the default zone annotation. I will revert this as it makes no sense here anyway.

**Review by toelke — APPROVED** [review]

**Pull request merged**